tfhe/src/typed_api/errors.rs

```diff
@@ -0,0 +1,165 @@
+use std::fmt::{Display, Formatter};
+
+/// Unwrap 'Extension' trait
+///
+/// The goal of this trait is to add a method similar to `unwrap` to `Result<T, E>`
+/// that uses the implementation of `Display` and not `Debug` as the
+/// message in the panic.
+pub trait UnwrapResultExt<T> {
+    fn unwrap_display(self) -> T;
+}
+
+impl<T, E> UnwrapResultExt<T> for Result<T, E>
+where
+    E: Display,
+{
+    #[track_caller]
+    fn unwrap_display(self) -> T {
+        match self {
+            Ok(t) => t,
+            Err(e) => panic!("{}", e),
+        }
+    }
+}
+
+/// Enum that lists types available
+///
+/// Mainly used to provide good errors.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Type {
+    #[cfg(feature = "boolean")]
+    FheBool,
+    #[cfg(feature = "shortint")]
+    FheUint2,
+    #[cfg(feature = "shortint")]
+    FheUint3,
+    #[cfg(feature = "shortint")]
+    FheUint4,
+    #[cfg(feature = "integer")]
+    FheUint8,
+    #[cfg(feature = "integer")]
+    FheUint10,
+    #[cfg(feature = "integer")]
+    FheUint12,
+    #[cfg(feature = "integer")]
+    FheUint14,
+    #[cfg(feature = "integer")]
+    FheUint16,
+    #[cfg(feature = "integer")]
+    FheUint256,
+}
+
+/// The server key of a given type was not initialized
+#[derive(Debug)]
+pub struct UninitializedServerKey(pub(crate) Type);
+
+impl Display for UninitializedServerKey {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "The server key for the type '{:?}' was not properly initialized",
```

Comment:
If we cant to tell which function to call, since each key has its own param, I would need to store some static object/ do a function that maps a type to the string that contains the name of the config function to call

Reply:
isn't there an "inintialize_server_keys" that's more or less always the same ?

Reply:
there is the `set_server_key` however if in your config you did not call `enable_default_u256` and that you try do operations on 256 bits, even if you call `set_server_key` you could have that error pop up.

To keep it simple, the message could be  `
 "
The server key for the type '{:?}' was not properly initialized
Did you forget to call `set_server_key` or initialize the type '{:?}" in the config ?
"